### ISSUE TITLE
`try` inside of `each` with a non-zero external command does not execute `catch`

### ISSUE THREAD

**Issue by Taywee** [nushell/nushell#12415]
### Describe the bug

I've found a specific circumstance where a `try` can throw an error but skip executing the `catch`.

Interestingly, it's only in the specific circumstances for me:

* The try block is inside of an `each`.
* The error is triggered by an external command.
* There are no statements inside the `each` after the `catch` block.

### How to reproduce

```nushell
> [1] | each { try { print false; ^false } catch { print true } }
false
╭───┬──╮
│ 0 │  │
╰───┴──╯
```

Note that the following things will prevent the bug from showing:

Making the error via a non-external command:

```nushell
> [1] | each { try { print false; error make {msg: foo} } catch { print true } }
false
true
╭────────────╮
│ empty list │
╰────────────╯
```

Having anything after the `catch`:

```nushell
> [1] | each { try { print false; ^false } catch { print true }; null }
false
true

╭────────────╮
│ empty list │
╰────────────╯
```

Having the `try` statement without an `each`:

```nushell
> try { print false; ^false } catch { print true }
false
true
```

### Expected behavior

I expect the `catch` to execute in a consistent way.

### Configuration

| key                | value                                                      |
| ------------------ | ---------------------------------------------------------- |
| version            | 0.92.1                                                     |
| branch             |                                                            |
| commit_hash        |                                                            |
| build_os           | linux-x86_64                                               |
| build_target       | x86_64-unknown-linux-gnu                                   |
| rust_version       | rustc 1.77.0 (aedd173a2 2024-03-17) (Fedora 1.77.0-1.fc39) |
| cargo_version      | cargo 1.77.0                                               |
| build_time         | 2024-04-04 00:00:00 +00:00                                 |
| build_rust_channel | release                                                    |
| allocator          | mimalloc                                                   |
| features           | default, sqlite, trash, which                              |
| installed_plugins  | gstat, bexpand                                             |

### Additional context

I discovered this while trying to do something like `glob submodules/* | par-each {|submodule| cd $submodule; try {git switch edge } catch {git switch main}}`, and I discovered that none of the catch blocks were actually running.

**Comment by weirdan** [comment]
With 0.110.0 both `false` and `true` are printed, and the list is empty, just like with internal errors:

```console
❯ [1] | each { try { print false; ^false } catch { print true } }
false
true
╭────────────╮
│ empty list │
╰────────────╯
```

**Comment by Taywee** [comment]
You're right.  It looks like it's been fixed since I opened this issue.  I forgot about this one.  Thanks for letting me know.
